File: frontend/src/lib/features/library/types/library.ts
export type LibraryItem = {
  id: number;
  filename: string;
  original_filename: string;
  file_path: string;
  markdown_path: string;
  imported_at: string;
  status: string;
  collection_status: 'inbox' | 'workspace' | 'library';
  project_id: string | null;
  title: string | null;
  authors: string | null;
  journal: string | null;
  conference: string | null;
  year: number | null;
  doi: string | null;
  abstract: string | null;
  keywords: string | null;
  publisher: string | null;
  source_url: string | null;
  metadata_source: string | null;
  metadata_confidence: string | null;
  metadata_updated_at: string | null;
  reading_progress: number;
  importance: string | null;
  priority: string | null;
  domain: string | null;
  method: string | null;
  difficulty: string | null;
  personal_tags: string | null;
};

export type LibraryListResponse = {
  items: LibraryItem[];
};

export type LibraryImportConflict = {
  reason: 'already_imported';
  message: string;
  existing_item: LibraryItem;
};

export type DuplicateStrategy = 'cancel' | 'replace' | 'keep_both';

export type PdfMetadataPreview = {
  title: string;
  authors: string;
  journal: string;
  conference: string;
  year: number | null;
  doi: string;
  abstract: string;
  keywords: string;
  publisher: string;
  source_url: string;
  metadata_source: string;
  metadata_confidence: string;
  reading_status: string;
  reading_progress: number;
  importance: string;
  priority: string;
  domain: string;
  method: string;
  difficulty: string;
  personal_tags: string;
  project_id: string;
};

File: backend/schemas/library.py
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel


class DuplicateStrategy(StrEnum):
    cancel = "cancel"
    replace = "replace"
    keep_both = "keep_both"


class LibraryItemResponse(BaseModel):
    id: int
    filename: str
    original_filename: str
    file_path: str
    markdown_path: str
    imported_at: datetime
    status: str
    collection_status: str = "inbox"
    project_id: str | None = None
    title: str | None = None
    authors: str | None = None
    journal: str | None = None
    conference: str | None = None
    year: int | None = None
    doi: str | None = None
    abstract: str | None = None
    keywords: str | None = None
    publisher: str | None = None
    source_url: str | None = None
    metadata_source: str | None = None
    metadata_confidence: str | None = None
    metadata_updated_at: datetime | None = None
    reading_progress: int = 0
    importance: str | None = None
    priority: str | None = None
    domain: str | None = None
    method: str | None = None
    difficulty: str | None = None
    personal_tags: str | None = None

    model_config = {"from_attributes": True}


class PdfMetadataPreviewResponse(BaseModel):
    title: str = ""
    authors: str = ""
    journal: str = ""
    conference: str = ""
    year: int | None = None
    doi: str = ""
    abstract: str = ""
    keywords: str = ""
    publisher: str = ""
    source_url: str = ""
    metadata_source: str = "pdf"
    metadata_confidence: str = "low"
    reading_status: str = "unread"
    reading_progress: int = 0
    importance: str = ""
    priority: str = "normal"
    domain: str = ""
    method: str = ""
    difficulty: str = ""
    personal_tags: str = ""
    project_id: str = ""


class LibraryImportConflictResponse(BaseModel):
    reason: str = "already_imported"
    message: str
    existing_item: LibraryItemResponse


class LibraryListResponse(BaseModel):
    items: list[LibraryItemResponse]
